session/pessimistic_test.go

```diff
@@ -1786,3 +1788,37 @@ func (s *testPessimisticSuite) TestSelectForUpdateWaitSeconds(c *C) {
 	tk4.MustExec("rollback")
 	tk5.MustExec("rollback")
 }
+
+func (s *testPessimisticSuite) TestSelectForUpdateConflictRetry(c *C) {
+	defer config.RestoreFunc()()
+	config.UpdateGlobal(func(conf *config.Config) {
+		conf.TiKVClient.EnableAsyncCommit = true
+	})
+
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("drop table if exists tk")
+	tk.MustExec("create table tk (c1 int primary key, c2 int)")
+	tk.MustExec("insert into tk values(1,1),(2,2)")
+	tk2 := testkit.NewTestKitWithInit(c, s.store)
+	tk3 := testkit.NewTestKitWithInit(c, s.store)
+
+	tk2.MustExec("begin pessimistic")
+	tk3.MustExec("begin pessimistic")
+	tk2.MustExec("update tk set c2 = c2 + 1 where c1 = 1")
+	tk3.MustExec("update tk set c2 = c2 + 1 where c2 = 2")
+	tsCh := make(chan uint64)
+	go func() {
+		tk3.MustExec("update tk set c2 = c2 + 1 where c1 = 1")
+		tk3.MustExec("commit")
+		lastTS, err := s.store.GetOracle().GetLowResolutionTimestamp(context.Background())
```

Comment:
Should we put this line above the previous line, to make it more clear that we get the ts after processing the pessimistic write conflict before the commit.

Reply:
Makes sence. And I also move `GetLowResolutionTimestamp` for tk2 before its commit, otherwise we cannot make sure it's earlier than tk3 obtains a new TS.